pkg/testgridanalysis/testidentification/ocp_variants.go

```diff
@@ -28,20 +30,22 @@ var (
 	ovirtRegex     = regexp.MustCompile(`(?i)-ovirt`)
 	ovnRegex       = regexp.MustCompile(`(?i)-ovn`)
 	// proxy jobs do not have a trailing -version segment
-	proxyRegex   = regexp.MustCompile(`(?i)-proxy`)
-	promoteRegex = regexp.MustCompile(`(?i)^promote-`)
-	ppc64leRegex = regexp.MustCompile(`(?i)-ppc64le`)
-	rtRegex      = regexp.MustCompile(`(?i)-rt`)
-	s390xRegex   = regexp.MustCompile(`(?i)-s390x`)
-	serialRegex  = regexp.MustCompile(`(?i)-serial`)
-	techpreview  = regexp.MustCompile(`(?i)-techpreview`)
-	upgradeRegex = regexp.MustCompile(`(?i)-upgrade`)
+	ppc64leRegex      = regexp.MustCompile(`(?i)-ppc64le`)
+	promoteRegex      = regexp.MustCompile(`(?i)^promote-`)
+	proxyRegex        = regexp.MustCompile(`(?i)-proxy`)
+	rtRegex           = regexp.MustCompile(`(?i)-rt`)
+	s390xRegex        = regexp.MustCompile(`(?i)-s390x`)
+	serialRegex       = regexp.MustCompile(`(?i)-serial`)
+	singleNodeRegex   = regexp.MustCompile(`(?i)-single-node`)
+	techpreview       = regexp.MustCompile(`(?i)-techpreview`)
+	upgradeMinorRegex = regexp.MustCompile(`(?i)(-\d+\.\d+-.*-.*-\d+\.\d+)|(-\d+\.\d+-minor)`)
```

Comment:
I think it works. We'll check if it's a minor upgrade, and then if not mark it as a micro:

https://regex101.com/r/it3rQD/1